### ISSUE TITLE
Extra SET calls

### ISSUE THREAD

**Issue by mik3y** [bernardopires/django-tenant-schemas#50]
First, thanks for this project.  Although I am not using it since I don't use postgres, I found it helpful when researching multi-tenancy options.

I believe I have the answer to one mystery [noted in the middleware](https://github.com/bcarneiro/django-tenant-schemas/blob/master/tenant_schemas/middleware.py#L14-L23).  There, each request gratuitously issues `connection.set_schema_to_public()` first, soon followed by `connection.set_tenant(..)`.

When using devserver, which is multithreaded by default (unless `--nothreading` is specified), I noticed a single thread only ever services a single `connection` object, ie, a new thread is spawned per request, and is discarded at the end of the request. 

However, other servers such as [gunicorn](http://gunicorn.org/) keep worker pools, meaning a long-lived thread will service distinct `connection` objects throughout its lifetime.

Since Django closes connections at the end of the request (look for `signals.request_finished` in `django.db.__init__`), it means the connection state is effectively reset after a request -- even though the _threadlocal_ data persists.  Thus the threadlocal data (applied search paths) is out-of-sync with the true connection state (default for new connection).

In summary, I believe you should attach some thread local state to the connection, not just to the thread.  Since connections are threadlocal in django unless `allow_thread_sharing` is set, it'd probably be safe to just set/test an attribute the connection (ie `cursor.connection.applied_tenant = tenant`).  (In fact, it shouldn't matter if the connection is shared or not, since the applied search path is connection state regardless).

This should also help you avoid spurious `SET` calls when the tenant has already been set on a given connection.

Hope this helps!
